### PR TITLE
Minor System.Collections.NonGeneric test cleanup

### PR DESCRIPTION
**Pull request by hughbe**
I was having a nostaligic through some of my first PRs to corefx, and checked out the System.Collections.NonGeneric tests
- Rename tests to remove the Test prefix
--> Is it worth removing the `static` keyword for these tests or not?
--> Sorry, this makes the diff a tad lengthy
- Add tests for the parameter name for `ArgumentExceptions`
- Cleans up some ctor and enumerator tests
- Adds some missing parameters to the production code (separate commit)

/cc @ianhays (I think you reviewed my original PR in this test lib)

### CODE REVIEW

File: src/System.Collections.NonGeneric/tests/CollectionBaseTests.cs [hunk 1]
```diff
@@ -241,16 +238,16 @@ public static void TestCopyTo()
...
         [Fact]
-        public static void TestCopyTo_Invalid()
+        public static void CopyTo_Invalid()
         {
             MyCollection collBase = CreateCollection(100);
             var fooArr = new Foo[100];
-            Assert.Throws<ArgumentOutOfRangeException>(() => collBase.CopyTo(fooArr, -1)); // Index < 0
-            Assert.Throws<ArgumentException>(() => collBase.CopyTo(fooArr, 50)); // Index + fooArray.Length > collBase.Count
+            Assert.Throws<ArgumentOutOfRangeException>("dstIndex", () => collBase.CopyTo(fooArr, -1)); // Index < 0
+            Assert.Throws<ArgumentException>("", () => collBase.CopyTo(fooArr, 50)); // Index + fooArray.Length > collBase.Count
```
Comment (ianhays) [inline src/System.Collections.NonGeneric/tests/CollectionBaseTests.cs]: string.empty?

Comment (hughbe) [inline src/System.Collections.NonGeneric/tests/CollectionBaseTests.cs]: I'm not sure if there's a preference between string.Empty and "" tests, but sure!

File: src/System.Collections.NonGeneric/tests/SortedListTests.cs [hunk 2]
```diff
@@ -1003,17 +989,17 @@ public static void TestGetValueList_CopyTo(int count, int index)
...
             {
                 IList values = sortList2.GetValueList();
-                Assert.Throws<ArgumentNullException>(() => values.CopyTo(null, 0)); // Array is null
-                Assert.Throws<ArgumentException>(() => values.CopyTo(new object[10, 10], 0)); // Array is multidimensional
+                Assert.Throws<ArgumentNullException>("dest", () => values.CopyTo(null, 0)); // Array is null
+                Assert.Throws<ArgumentException>("array", () => values.CopyTo(new object[10, 10], 0)); // Array is multidimensional
 
-                Assert.Throws<ArgumentOutOfRangeException>(() => values.CopyTo(new object[100], -1)); // Index < 0
-                Assert.Throws<ArgumentException>(() => values.CopyTo(new object[150], 51)); // Index + list.Count > array.Count
+                Assert.Throws<ArgumentOutOfRangeException>("dstIndex", () => values.CopyTo(new object[100], -1)); // Index < 0
```
Comment (joshfree) [inline src/System.Collections.NonGeneric/tests/SortedListTests.cs]: nit: consider "destIndex" for all occurrences

**Comment by joshfree** [comment]
overall LGTM.  Thanks @hughbe

File: src/System.Collections.NonGeneric/tests/SortedListTests.cs (hunk 2 above)
Comment (hughbe) [inline src/System.Collections.NonGeneric/tests/SortedListTests.cs]: Unfortunately, I don't think I have control of the parameter name, as the implementation of CopyTo across the lib doesn't perform any argument validation. 

For example, the Param name here is "index" but the thrown Param is "dstIndex"

I considered adding custom argument validation in these situations, but I'm not certain about the Perf effect or changing legacy code

File: src/System.Collections.NonGeneric/tests/HashtableTests.cs [hunk 3]
```diff
@@ -747,27 +749,27 @@ public static void TestCopyTo_EmptyHashtable()
...
                 Assert.Throws<InvalidCastException>(() => hash2.CopyTo(new object[10][], 0)); // Array is invalid
 
-                Assert.Throws<ArgumentOutOfRangeException>(() => hash2.CopyTo(new object[10], -1)); // Index < 0
-                Assert.Throws<ArgumentException>(() => hash2.CopyTo(new object[9], 0)); // Hash.Count + index > array.Length
-                Assert.Throws<ArgumentException>(() => hash2.CopyTo(new object[11], 2)); // Hash.Count + index > array.Length
-                Assert.Throws<ArgumentException>(() => hash2.CopyTo(new object[0], 0)); // Hash.Count + index > array.Length
+                Assert.Throws<ArgumentOutOfRangeException>("arrayIndex", () => hash2.CopyTo(new object[10], -1)); // Index < 0
+                Assert.Throws<ArgumentException>(null,() => hash2.CopyTo(new object[9], 0)); // Hash.Count + index > array.Length
+                Assert.Throws<ArgumentException>(null, () => hash2.CopyTo(new object[11], 2)); // Hash.Count + index > array.Length
+                Assert.Throws<ArgumentException>(null, () => hash2.CopyTo(new object[0], 0)); // Hash.Count + index > array.Length
```
Comment (ianhays) [inline src/System.Collections.NonGeneric/tests/HashtableTests.cs]: I'm curious what the benefit is to adding the parameter name checks when they're null.

File: src/System.Collections.NonGeneric/tests/HashtableTests.cs [hunk 4]
```diff
@@ -1415,36 +1414,20 @@ public override int GetHashCode()
...
 
         private class Foo
         {
-            private string _stringValue = "Hello World";
-            public string StringValue
-            {
-                get { return _stringValue; }
-                set { _stringValue = value; }
-            }
+            public string StringValue { get; set; } = "Hello World";
```
Comment (ianhays) [inline src/System.Collections.NonGeneric/tests/HashtableTests.cs]: Interesting syntax. Haven't ever seen this before.

File: src/System.Collections.NonGeneric/tests/HashtableTests.cs (hunk 3 above)
Comment (hughbe) [inline src/System.Collections.NonGeneric/tests/HashtableTests.cs]: If they ever change, then they are flagged up and can be updated

File: src/System.Collections.NonGeneric/tests/HashtableTests.cs (hunk 4 above)
Comment (hughbe) [inline src/System.Collections.NonGeneric/tests/HashtableTests.cs]: This is new c# 6 stuff - its quite neat in reducing boilerplate and writing clearer code

File: src/System.Collections.NonGeneric/tests/StackTests.cs [hunk 5]
```diff
@@ -485,20 +472,20 @@ public class Stack_SyncRootTests
...
-        private const int iNumberOfElements = 100;
+        private const int NumberOfElements = 100;
 
         [Fact]
-        public void TestGetSyncRootBasic()
+        public void GetSyncRootBasic()
         {
             // Testing SyncRoot is not as simple as its implementation looks like. This is the working
-            //scenrio we have in mind.
+            // scenrio we have in mind.
```
Comment (ianhays) [inline src/System.Collections.NonGeneric/tests/StackTests.cs]: nit: scenario

**Comment by ianhays** [comment]
Couple of nits/comments but otherwise LGTM.

**Comment by hughbe** [comment]
Thanks both for your comments and review - I addressed them (apart from the null param one and the dstIndex - I left comments explaining why I didn't address 'em. If you have a differing opinion, let me know and I'll fix them)

**Comment by stephentoub** [comment]
LGTM

**Pull request merged**